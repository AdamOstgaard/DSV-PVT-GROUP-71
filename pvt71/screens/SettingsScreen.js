import React from "react";
import { StyleSheet, Text, View, ScrollView} from "react-native";
import AppSingleButton from "../components/AppSingleButton";
import { getStatusBarHeight } from 'react-native-status-bar-height';

export default class SettingsScreen extends React.Component {
  render() {
    return (
      <View style = {styles.container}>
        <AppSingleButton
          style={styles.topButton}
          textStyle={styles.textTop}
          title ="Tillbaka"
            onPress={() =>
              this.props.navigation.navigate("HomeScreen",{})}
        ></AppSingleButton>
        <ScrollView style = {styles.scrollContainer}>

          <Text style ={styles.category}>Inställningar</Text>

          <AppSingleButton
          style={styles.linkButton}
          textStyle ={styles.text}
          title="Redigera tidsintervall"
            onPress={() =>
              this.props.navigation.navigate("TimerSettingsScreen",{})}
          ></AppSingleButton>

          <AppSingleButton
          style={styles.linkButton}
          textStyle ={styles.text}
          title="Redigera varningsintervall"
            onPress={() => 
              this.props.navigation.navigate("WarningSettingScreen",{})}
          ></AppSingleButton>

          <AppSingleButton
          style={styles.linkButton}
          textStyle ={styles.text}
          title="Redigera sömntider"
            onPress={() => 
              this.props.navigation.navigate("SleepTimeSettingsScreen",{})}
          ></AppSingleButton>

          <AppSingleButton
          style={styles.linkButton}
          textStyle ={styles.text}
          title="Redigera kontaktperson"
            onPress={() =>
              this.props.navigation.navigate("HomeScreen",{})}
          ></AppSingleButton>

          <AppSingleButton
          style={styles.linkButton}
          textStyle ={styles.text}
          title="Pausa applikationen"
            onPress={() =>
              this.props.navigation.navigate("HomeScreen",{})}
          ></AppSingleButton>
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginTop: getStatusBarHeight(),
    flex: 1,
    backgroundColor: "#fff",
  },
  scrollContainer: {},
  topButton: {
    width:"100%",
  },
  category: {
    fontWeight: 'bold',
    fontSize: 32,
    borderBottomColor: "#000",
    borderBottomWidth: 0.5,
    marginTop: "5%"
  },
  linkButton: {
    fontSize: 20,
    color: "#000",
    backgroundColor: "#fff",
    width:"100%",
    borderBottomColor: "#000",
    borderBottomWidth: 0.5,
  },
  text: {
    fontSize: 20,
    color: "black",
    paddingRight: 10,
    paddingTop: 16,
    textAlign: 'left',
  },
  textTop: {
    fontSize: 30,
    color: "#fff",
    paddingRight: 10,
    paddingTop: 8,
    textAlign: 'left'
  }
});
