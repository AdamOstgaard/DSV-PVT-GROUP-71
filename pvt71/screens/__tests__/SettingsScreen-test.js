import "react-native";
import React from "react";
import SettingsScreen  from "../SettingsScreen";
import renderer from "react-test-renderer";

describe("Wizard Verify Contact Screen tests", () => {
    describe("Snapshot test", () => {
        it("renders correctly", () => {
        const tree = renderer.create(<SettingsScreen />).toJSON();
        expect(tree).toMatchSnapshot();
        });
    });
});