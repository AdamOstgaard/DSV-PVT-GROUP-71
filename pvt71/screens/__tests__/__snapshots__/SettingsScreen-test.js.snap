// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting Screen tests Snapshot test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "marginTop": 20,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0d6f73",
        "height": 60,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
          "paddingRight": 10,
          "paddingTop": 8,
          "textAlign": "left",
        }
      }
    >
      Tillbaka
    </Text>
  </View>
</View>
`;

exports[`Wizard Verify Contact Screen tests Snapshot test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Settings Screen
  </Text>
</View>
`;
